reporter/samples/samples.go

```diff
@@ -53,6 +53,13 @@ type TraceAndMetaKey struct {
 	ExtraMeta any
 }
 
+// TraceEventsTree stores samples and their related metadata in a tree-like
+// structure optimized for the OTel Profiling protocol representation.
+type TraceEventsTree map[ContainerID]map[libpf.Origin]KeyToEventMapping
```

Comment:
`TraceEventsTree` is very generic.
Since it's a grouping by container, maybe name it more specific like `TraceEventsByContainer` or similar.

Reply:
These indirections in namings will make the code less readable and harder to follow. In particular if every nested level gets their own type name.
It does make sense to use dedicated subnames for some part of nested maps, if these parts are used outside of the scope of the tree. But I think this is not the case here.